src/helpers/JpegXL.cpp

```diff
@@ -0,0 +1,129 @@
+#include "JpegXL.hpp"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <filesystem>
+#include <jxl/decode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+
+cairo_surface_t* JXL::createSurfaceFromJXL(const std::string& path) {
+
+    if (!std::filesystem::exists(path)) {
+        Debug::log(ERR, "createSurfaceFromJXL: file doesn't exist??");
+        exit(1);
+    }
+
+    void*       imageRawData;
+
+    struct stat fileInfo = {};
+
+    const auto  FD = open(path.c_str(), O_RDONLY);
+
+    fstat(FD, &fileInfo);
+
+    imageRawData = malloc(fileInfo.st_size);
+
+    read(FD, imageRawData, fileInfo.st_size);
+
+    close(FD);
```

Comment:
wait a sec, why are we doing all this C style? `std::ifstream` can read binary, and is more memory safe. Then, just throw it into a `std::vector<uint8_t>` that you first `.resize()` to the filesize.

Reply:
completely agree, that's not how I usually do this kind of stuff, but I was checking all the other image loaders, and just did it in the same style.

Reply:
If you'd like, we can merge this as is, and I can make another PR that modernizes the loading part using streams and such.

Reply:
If you'd like, we can merge this as is, and I can make another PR that modernizes the loading part of all image loaders using streams and such.

Reply:
I'd rather do it here the C++ way in this MR and then work on others

Reply:
Ok, give me a sec!

Reply:
@vaxerski how about now?